src/Facets/EcoFacet.sol

```diff
@@ -0,0 +1,256 @@
+    /// @param _bridgeData Bridge data containing core parameters
+    /// @param _ecoData Eco-specific parameters for the bridge
+    function startBridgeTokensViaEco(
+        ILiFi.BridgeData memory _bridgeData,
+        EcoData calldata _ecoData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        validateBridgeData(_bridgeData)
+        doesNotContainSourceSwaps(_bridgeData)
+    {
+        // Validate eco-specific data before depositing
+        _validateEcoData(_bridgeData, _ecoData);
+
+        // For ERC20, we need to deposit the full amount including reward
+        uint256 depositAmount = _bridgeData.minAmount;
+        if (!LibAsset.isNativeAsset(_bridgeData.sendingAssetId)) {
+            depositAmount += _ecoData.solverReward;
+        }
+
+        LibAsset.depositAsset(_bridgeData.sendingAssetId, depositAmount);
+
+        _startBridge(_bridgeData, _ecoData);
+    }
+
+    /// @notice Swaps and bridges tokens via Eco Protocol
+    /// @param _bridgeData Bridge data containing core parameters
+    /// @param _swapData Array of swap data for source swaps
+    /// @param _ecoData Eco-specific parameters for the bridge
+    function swapAndStartBridgeTokensViaEco(
+        ILiFi.BridgeData memory _bridgeData,
+        LibSwap.SwapData[] calldata _swapData,
+        EcoData calldata _ecoData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        containsSourceSwaps(_bridgeData)
+        validateBridgeData(_bridgeData)
+    {
+        // Validate eco-specific data before swapping
+        _validateEcoData(_bridgeData, _ecoData);
+
+        // For ERC20 tokens, we need to reserve the solver reward from the swapped amount
+        // Only pass native fee reservation if the final asset is native
+        uint256 nativeFeeAmount = LibAsset.isNativeAsset(
+            _bridgeData.sendingAssetId
+        )
+            ? _ecoData.solverReward
+            : 0;
+
+        _bridgeData.minAmount = _depositAndSwap(
+            _bridgeData.transactionId,
+            _bridgeData.minAmount,
+            _swapData,
+            payable(msg.sender),
+            nativeFeeAmount
+        );
+
+        // For ERC20 tokens, the swap result includes the solver reward
+        // We need to subtract it to get the actual bridge amount
+        if (!LibAsset.isNativeAsset(_bridgeData.sendingAssetId)) {
+            _bridgeData.minAmount =
+                _bridgeData.minAmount -
+                _ecoData.solverReward;
+        }
+
+        _startBridge(_bridgeData, _ecoData);
+    }
+
+    /// Internal Methods ///
+
+
+    function _getEcoChainId(
+        uint256 _lifiChainId
+    ) private pure returns (uint64) {
+        // Map LiFi Tron chain ID to Eco protocol Tron chain ID
+        if (_lifiChainId == LIFI_CHAIN_ID_TRON) {
+            return 728126428; // Eco protocol's Tron chain ID
```

Comment:
would recommend to store this as immutable variable and use the variable here (as well as in tests). 
This will most likely be raised in the audit, too.